### PR TITLE
feat: send cannot execute event to amplifier

### PR DESCRIPTION
**Pull request by frenzox**
**Describe the changes**
Sending feedback to the amplifier in case of failure to execute a message.

**Related Issue(s)**

Closes https://github.com/eigerco/solana-axelar-internal/issues/504

**Checklist**

- [ ] Tests have been added/updated for the changes.
- [x] Documentation has been updated for the changes (if applicable).
- [x] The code follows Rust's style guidelines.

### CODE REVIEW

File: crates/solana-gateway-task-processor/src/component.rs [hunk 1]
```diff
@@ -140,26 +153,44 @@ struct ConfigMetadata {
...
         GatewayTx(task) => {
             gateway_tx_task(task, gateway_root_pda, signer, solana_rpc_client, keypair).await?;
         }
         Execute(task) => {
+            let source_chain = task.message.source_chain.clone();
+            let message_id = task.message.message_id.clone();
+
             // communicate with the destination program
-            execute_task(task, metadata, signer, solana_rpc_client, keypair)
+            if let Err(error) = execute_task(task, metadata, signer, solana_rpc_client, keypair)
```
Comment (frenzox) [inline crates/solana-gateway-task-processor/src/component.rs:176]: We need to update `send_tx` to return the error even when not irrecoverable and parse the errors here to return the appropriate event.

File: crates/effective-tx-sender/src/lib.rs [hunk 2]
```diff
@@ -114,10 +114,28 @@ impl<'a> EffectiveTxSender<'a, Evaluated> {
...
+            Err(err) if err.get_transaction_error().is_some() => {
+                let signature = tx
+                    .signatures
+                    .first()
+                    .expect("Signed transaction should have a signature");
+
+                return Err(ComputeBudgetError::TransactionError {
+                    source: err
+                        .get_transaction_error()
+                        .expect("Value shouldn't disapper after is_some returns true"),
```
Comment (frenzox) [inline crates/effective-tx-sender/src/lib.rs:132]:
```suggestion
                        .expect("Value shouldn't disappear after is_some returns true"),
```

**Review by roberts-pumpurs — DISMISSED** [review]
overall lgtm but you will need to alter the `fetch_logs` fn before merging. Because at this point in time the `solana_listener` component expects to only work with txs that succeeded.

File: crates/solana-gateway-task-processor/src/component.rs [hunk 3]
```diff
@@ -140,26 +155,67 @@ struct ConfigMetadata {
...
                 .in_current_span()
-                .await?;
+                .await
+            {
+                let event = match error.downcast_ref::<ComputeBudgetError>() {
+                    Some(&ComputeBudgetError::TransactionError {
+                        source: ref _source,
+                        signature,
+                    }) => {
+                        let tx = solana_listener::fetch_logs(signature, solana_rpc_client).await?;
```
[the remark this answers is not in the document]
Comment (frenzox) [reply crates/solana-gateway-task-processor/src/component.rs:188]: Ohh, good catch. I'll do that. Thanks!

[the remark this answers is not in the document]
Comment (frenzox) [reply crates/solana-gateway-task-processor/src/component.rs]: Updated, let me know if you think that's good enough

[the remark this answers is not in the document]
Comment (frenzox) [reply crates/solana-gateway-task-processor/src/component.rs]: Ok, I realised `fetch_logs` also fails when the logs are not available and here we're not really interested in the logs but in the fees and perhaps the block_time. Thus I just updated now to use the get_transaction rpc directly.

**Review by roberts-pumpurs — APPROVED** [review]

**Pull request merged**